File: year2/systems-design-implementation/Lab5/SwimModel/src/main/java/validator/ParticipantValidator.java
package validator;

import domain.Participant;

public class ParticipantValidator implements Validator<Participant>{
    public void validate(Participant participant) throws Exception {
        if(participant.getNume().equals("")){
            throw new Exception("Nume invalid");
        }
        if(participant.getVarsta()<=0){
            throw new Exception("Varsta invalida");
        }
    }
}

File: year2/systems-design-implementation/Lab4/Lab4Sharp/Domain/Validator/ParticipantValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab2Sharp.Domain.Validator
{
    public class ParticipantValidator: Validator<Participant>
    {
        public void validate(Participant participant)
        {
            if (participant.nume == "")
            {
                throw new Exception("Nume invalid");
            }
            if (participant.varsta <= 0)
            {
                throw new Exception("Varsta invalida");
            }
        }
    }
}
